controllers/util/secrets_util.go

```diff
@@ -54,12 +72,45 @@ type SecretsUtil struct {
 
 func GeneratePolicyResourceNames(
 	secret string,
+	format TargetSecretFormat,
 ) (policyName, plBindingName, plRuleName, configPolicyName string) {
-	// policyName is the same as secret name, to retain name length restrictions
-	return secret,
-		fmt.Sprintf(secretResourceNameFormat, secretPlBindingBaseName, secret),
-		fmt.Sprintf(secretResourceNameFormat, secretPlRuleBaseName, secret),
-		fmt.Sprintf(secretResourceNameFormat, secretConfigPolicyBaseName, secret)
+	var pName string
+
+	switch format {
+	case SecretFormatRamen:
+		pName = ramenFormatPrefix + secret
+	case SecretFormatVelero:
+		pName = veleroFormatPrefix + secret
+	}
```

Comment:
The only reasonable thing to do is assert (panic), so that devs can catch a new format that is not handled in the future, or return and handle errors across all the functions using these internal routines.

Currently not adding a panic here for assertion as this is quite controlled in use.

Reply:
While the call is controlled in use, adding a `default` case would help prevent unintended changes to the function argument, especially since a `string` argument can be easily modified without notice. Additionally, `VSCode`'s autocomplete feature can occasionally introduce unexpected changes found only during testing.